### PR TITLE
PLAT-751: add SMVCachedMemoryRequest

### PR DESCRIPTION
**Pull request by ElSacramento**
simple SMVCacherMemoryRequest

### CODE REVIEW

File: ledger-core/virtual/handlers/vcachedmemoryrequest.go [hunk 1]
```diff
@@ -6,35 +6,131 @@
...
+	injector.MustInject(&s.messageSender)
+}
+
 func (s *SMVCachedMemoryRequest) GetStateMachineDeclaration() smachine.StateMachineDeclaration {
 	return dSMVCachedMemoryRequestInstance
 }
+
+func (s *SMVCachedMemoryRequest) migrationDefault(ctx smachine.MigrationContext) smachine.StateUpdate {
+	ctx.Log().Trace("stop processing VCachedMemoryRequest since pulse was changed")
+	return ctx.Stop()
```
Comment (ruz) [inline ledger-core/virtual/handlers/vcachedmemoryrequest.go:72]: this is wrong migration, this request probably shouldn't have migration and continue working.

File: ledger-core/virtual/handlers/vcachedmemoryrequest.go [hunk 2]
```diff
@@ -6,35 +6,131 @@
...
+}
+
 func (s *SMVCachedMemoryRequest) Init(ctx smachine.InitializationContext) smachine.StateUpdate {
+	if s.pulseSlot.State() != conveyor.Present {
+		ctx.Log().Trace("stop processing VCachedMemoryRequest since we are not in present pulse")
```
Comment (ruz) [inline ledger-core/virtual/handlers/vcachedmemoryrequest.go:77]: no need in this check

File: ledger-core/virtual/handlers/vcachedmemoryrequest.go [hunk 3]
```diff
@@ -6,35 +6,131 @@
...
+		return ctx.Stop()
+	}
+	ctx.SetDefaultMigration(s.migrationDefault)
+	return ctx.Jump(s.getMemory)
+}
+
+func (s *SMVCachedMemoryRequest) getMemory(ctx smachine.ExecutionContext) smachine.StateUpdate {
```
Comment (ruz) [inline ledger-core/virtual/handlers/vcachedmemoryrequest.go:84]: we agreed to name all steps with "step" prefix

File: ledger-core/virtual/handlers/vcachedmemoryrequest.go (hunk 2 above)
Comment (ElSacramento) [reply ledger-core/virtual/handlers/vcachedmemoryrequest.go]: removed

File: ledger-core/virtual/handlers/vcachedmemoryrequest.go (hunk 1 above)
Comment (ElSacramento) [reply ledger-core/virtual/handlers/vcachedmemoryrequest.go]: removed

File: ledger-core/virtual/handlers/vcachedmemoryrequest.go (hunk 3 above)
Comment (ElSacramento) [reply ledger-core/virtual/handlers/vcachedmemoryrequest.go]: fixed

File: ledger-core/virtual/handlers/vcachedmemoryrequest.go [hunk 4]
```diff
@@ -6,35 +6,131 @@
...
+	}
+	return ctx.Jump(s.buildResult)
+}
+
+func (s *SMVCachedMemoryRequest) buildResult(ctx smachine.ExecutionContext) smachine.StateUpdate {
+	if s.object.HeadRef().IsEmpty() {
+		s.response = &payload.VCachedMemoryResponse{
+			Object:     s.Payload.Object,
+			StateID:    s.Payload.StateID,
+			CallStatus: payload.CachedMemoryStateMissing,
```
Comment (ruz) [inline ledger-core/virtual/handlers/vcachedmemoryrequest.go:95]: I think this should be unknown.

Comment (ElSacramento) [reply ledger-core/virtual/handlers/vcachedmemoryrequest.go:95]: What descriptor will be considered as Missing? with empty memory?

Comment (ruz) [reply ledger-core/virtual/handlers/vcachedmemoryrequest.go:95]: not with empty memory. Read documentation: unknown - "I don't have, but not sure about others", missing - "not existent in the whole system".

Comment (ElSacramento) [reply ledger-core/virtual/handlers/vcachedmemoryrequest.go]: fixed

**Review by ruz — APPROVED** [review]

**Review by ita-sammann — APPROVED** [review]

**Pull request merged**